### PR TITLE
Remove fee-rates periodic syncing functionality (feature)

### LINKED ISSUE #16: Refactor fee-rates syncing and fetching functionality
- Remove fee-rates periodic syncing functionality. Fee-rates should be provided only on demand.
- Add fallback provider,  using in memory storage with 5 min interval (expiration)

### PR DESCRIPTION
**Pull request by nk-the-crazy**
- Remove fee-rates periodic syncing feature
- Add fallback fee-rates provider with custom interval based on Coins.
- Close #16

### CODE REVIEW

File: feeratekit/src/main/kotlin/io/horizontalsystems/feeratekit/model/FeeRate.kt [hunk 1]
```diff
@@ -1,82 +1,80 @@
...
+    ETHEREUM("ETH", 3 * 60);
 
     companion object {
         fun getCoinByCode(code: String): Coin? {
-            return values().find { coin -> coin.code.equals(code)}
+            return values().find { coin -> coin.code.equals(code) }
         }
     }
 
     fun defaultRate(): FeeRate {
```
Comment (mNizhurin) [inline feeratekit/src/main/kotlin/io/horizontalsystems/feeratekit/model/FeeRate.kt:37]: We don't need default rates anymore?

File: feeratekit/src/main/kotlin/io/horizontalsystems/feeratekit/model/FeeRate.kt [hunk 2]
```diff
@@ -1,82 +1,80 @@
...
 }
 
-enum class Coin(val code: String) {
-    BITCOIN("BTC"),
-    BITCOIN_CASH("BCH"),
-    DASH("DASH"),
-    ETHEREUM("ETH");
+enum class Coin(
+        val code: String,
+        val fallbackDataExpiration: Int) {
```
Comment (mNizhurin) [inline feeratekit/src/main/kotlin/io/horizontalsystems/feeratekit/model/FeeRate.kt:24]: I think better place for expiration time would be in the BaseFeeRateProvider. Coin knowing about expiration time of FeeRate looks strange to me.

**Review by mNizhurin — CHANGES_REQUESTED** [review]

File: app/src/main/java/io/horizontalsystems/feeratekit/demo/MainViewModel.kt [hunk 3]
```diff
@@ -4,14 +4,18 @@ import androidx.lifecycle.MutableLiveData
...
-import io.horizontalsystems.feeratekit.providers.FeeProviderConfig
+import io.horizontalsystems.feeratekit.model.Coin
+import io.horizontalsystems.feeratekit.model.FeeProviderConfig
+import io.reactivex.schedulers.Schedulers
+import io.reactivex.disposables.CompositeDisposable
+
 
 class MainViewModel : ViewModel(), FeeRateKit.Listener {
-    val rates = MutableLiveData<List<FeeRate>>()
+    val rates = MutableLiveData<FeeRate>()
```
Comment (rafaelekol) [inline app/src/main/java/io/horizontalsystems/feeratekit/demo/MainViewModel.kt]: Maybe we should better rename `rates` to `feeRate`

File: app/src/main/java/io/horizontalsystems/feeratekit/demo/MainViewModel.kt [hunk 4]
```diff
@@ -23,21 +27,25 @@ class MainViewModel : ViewModel(), FeeRateKit.Listener {
...
+                FeeRate(Coin.BITCOIN,0,0,0,0,0,0,0)
+            }
+            .subscribeOn(Schedulers.io())
+            .subscribe() { t -> rates.postValue(t) })
+        compositeDisposable.add(feeRateKit.getRate("ETH")
+            .subscribeOn(Schedulers.io())
+            .subscribe() { t -> rates.postValue(t) })
+        compositeDisposable.add(feeRateKit.getRate("BCH")
+            .subscribeOn(Schedulers.io())
+            .subscribe() { t -> rates.postValue(t) })
```
Comment (rafaelekol) [inline app/src/main/java/io/horizontalsystems/feeratekit/demo/MainViewModel.kt]: Can we split disposables with new lines, for better reading

File: feeratekit/src/main/kotlin/io/horizontalsystems/feeratekit/storage/EnumConverter.kt [hunk 5]
```diff
@@ -1,15 +1,12 @@
...
 class EnumConverter {
-    @TypeConverter
     fun fromCoin(coin: Coin): String {
         return coin.code
     }
 
-    @TypeConverter
     fun toCoin(code: String): Coin {
         return Coin.values().first { it.code == code }
     }
```
Comment (rafaelekol) [inline feeratekit/src/main/kotlin/io/horizontalsystems/feeratekit/storage/EnumConverter.kt]: we can delete this class, if it's not used anymore

File: feeratekit/src/main/kotlin/io/horizontalsystems/feeratekit/providers/BaseFeeRateProvider.kt [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+                }
+                ?.doOnSuccess {
+                    rate -> storeToFallback(rate)
+                }
+                ?: Single.just(coin.defaultRate())
+    }
+
+    private fun provideFromFallback(): FeeRate? {
+
+        storage.getFeeRate(this.coin)?.let {
```
Comment (omurovch) [inline feeratekit/src/main/kotlin/io/horizontalsystems/feeratekit/providers/BaseFeeRateProvider.kt]: return can be moved to here

File: feeratekit/build.gradle [hunk 7]
```diff
@@ -33,22 +33,18 @@ dependencies {
     implementation fileTree(dir: 'libs', include: ['*.jar'])
     implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlin_version"
     implementation 'androidx.appcompat:appcompat:1.1.0-alpha04'
+    implementation 'org.jetbrains.kotlinx:kotlinx-coroutines-core:1.3.0'
```
Comment (omurovch) [inline feeratekit/build.gradle]: let's remove coroutine dependencies

**Review by omurovch — APPROVED** [review]

**Pull request merged**